### PR TITLE
Dynamically load libpulse.so.0 and libasound.so.1 on Linux

### LINKED ISSUE #20978: Load PulseAudio/ALSA dynamically (dlopen) on Linux / *BSD
**Godot version:**
Current master (51a0a381), but could be backported to other stable branches.

**OS/device including version:**
"X11" platform / any freedesktop.org OS (Linux distros, OpenBSD, FreeBSD).

**Issue description:**
For the X11 platform, we currently link the PulseAudio library (`libpulse.so.0`, `libpulsecommon-10.0.so`) directly in official binaries, which means that our binaries can't be used on systems without PulseAudio (especially relevant for Arch or Gentoo users, as well as any user with a strong aversion for PulseAudio or any Poettering endeavour).

This means that to support such users, game developers need to provide a Linux binary compiled with `pulseaudio=no`, which causes a fallback to ALSA (`libasound.so.2`).

```
[akien@cauldron godot.git (master)]$ ldd ~/tmp/godot/stable/Godot_v3.0.6-stable_x11.64 | grep pulse
        libpulse.so.0 => /lib64/libpulse.so.0 (0x00007f345cdf1000)
        libpulsecommon-10.0.so => /usr/lib64/pulseaudio/libpulsecommon-10.0.so (0x00007f345b6bf000)
[akien@cauldron godot.git (master)]$ ldd ~/tmp/godot/stable/Godot_v3.0.6-stable_x11.64 | grep asound
        libasound.so.2 => /lib64/libasound.so.2 (0x00007fbe43c9f000)
```

It would be better if we loaded these shared libraries dynamically with `dlopen()` instead of linking them upfront, allowing to run Godot even if they are missing (eventually defaulting to the Dummy audio driver if both are missing).

### PR DESCRIPTION
**Pull request by hpvb**
By generating stubs using https://github.com/hpvb/dynload-wrapper we
can dynamically load libpulse and libasound on systems where it is available.
Both are still a build-time requirement but no longer a run-time dependency.
    
For maintenance purposes the wrappers should not need to be re-generated
unless we want to bump pulse or asound to an incompatible version. It is
unlikely we will want to do this any time soon.
    
This closes #20978

### CODE REVIEW

**Review by reduz — APPROVED** [review]

File: drivers/pulseaudio/SCsub [hunk 1]
```diff
@@ -2,4 +2,6 @@
 
 Import("env")
 
+if "pulseaudio" in env:
```
Comment (akien-mga) [inline drivers/pulseaudio/SCsub:5]: That would include pulseaudio if you pass `pulseaudio=no` too. You should also check `and env["pulseaudio"]`.

Comment (hpvb) [reply drivers/pulseaudio/SCsub]: Done, thanks!

**Review by akien-mga — APPROVED** [review]

File: platform/linuxbsd/detect.py [hunk 2]
```diff
@@ -310,17 +310,16 @@ def configure(env):
 
     if os.system("pkg-config --exists alsa") == 0:  # 0 means found
         print("Enabling ALSA")
+        env["alsa"] = True
```
Comment (akien-mga) [inline platform/linuxbsd/detect.py:313]: Should we make it an option like pulseaudio so that users can choose not to compile alsa support in?

Comment (akien-mga) [reply platform/linuxbsd/detect.py:313]: Though that's maybe outside the scope of this PR. I suspect that alsamidi might fail compile if we disable alsa.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**